### ISSUE TITLE
Error adding bullets

### ISSUE THREAD

**Issue by jmpmscorp** [amcharts/amcharts4#1118]
Hi.

I'm using amcharts in an Angular 7 app. My app was working fine with v4.2.2. Since I have upgraded to v4.2.6, charts stop working. It seems to be related with bullets in my XY chart lineseries. If I comment or delete bullets code, chart works as expected. As I have written, with v4.2.2 works perfectly. I have tested with 4.2.5, 4.2.6 and 4.3.2 but the results are the same.

My component code is:

```typescript
@Component({
  selector: 'aquacheck-graph',
  templateUrl: './aquacheck-graph.component.html',
  styleUrls: ['./aquacheck-graph.component.scss']
})
export class AquacheckGraphComponent implements OnInit, OnDestroy {
  
  /* Rest of variables */
  private _chart: am4charts.XYChart;
  private _humAxis: am4charts.ValueAxis;

  private _retrievingDataSubject = new BehaviorSubject<boolean>(false);

  get retrievingData(): Observable<boolean> {
    return this._retrievingDataSubject.asObservable();
  }

  constructor(
    private ngZone: NgZone,
    private datosSondasService: DatosSondaService
  ) { }


  public searchSoil(): void {
    const searchParams = new HttpParams( {fromObject: {
      fechaInicio: format(this.startDate.value, 'yyyy-MM-dd'),
      fechaFin: format(this.endDate.value, 'yyyy-MM-dd')
    }}); 

    this._retrievingDataSubject.next(true);
    this.datosSondasService.getSuelo(parseInt(this.idFinca), parseInt(this.idNode), searchParams).subscribe(
      data => {
        this.ngZone.runOutsideAngular( () => {
          if(this._chart) {
            this._chart.dispose();
          }
          this._createChart(data);
        })
        
        this._retrievingDataSubject.next(false);
      },
      error => {
        console.error(error);
        this._retrievingDataSubject.next(false);
      }
    )
  }

  private _createChart( data: DatosSonda[] ): void {
    let chart = am4core.create("aquacheck-chart-div", am4charts.XYChart);

      chart.paddingRight = 20;

      let dateAxis = chart.xAxes.push(new am4charts.DateAxis());
      dateAxis.renderer.grid.template.location = 0;
      dateAxis.dateFormatter.inputDateFormat = 'i';
      dateAxis.dateFormats.setKey('hour', 'd MMM, HH:mm');
      dateAxis.periodChangeDateFormats.setKey('hour', 'd MMM, HH:mm');
      dateAxis.tooltipDateFormat = 'd MMM, HH:mm' 

      let humValueAxis = chart.yAxes.push(new am4charts.ValueAxis());
      humValueAxis.tooltip.disabled = true;
      humValueAxis.renderer.minWidth = 35;
      humValueAxis.min=0;
      humValueAxis.renderer.line.strokeOpacity = 1;
      humValueAxis.renderer.line.strokeWidth = 2;
      humValueAxis.renderer.line.stroke = am4core.color('blue');
      this._humAxis = humValueAxis;


      let tempValueAxis = chart.yAxes.push(new am4charts.ValueAxis());
      tempValueAxis.tooltip.disabled = true;
      tempValueAxis.renderer.minWidth = 35;
      tempValueAxis.renderer.opposite = true;
      tempValueAxis.renderer.line.strokeOpacity = 1;
      tempValueAxis.renderer.line.strokeWidth = 2;
      tempValueAxis.renderer.line.stroke = am4core.color('red');
      tempValueAxis.hideUnusedDataItems();
      
      chart.cursor = new am4charts.XYCursor();
      chart.cursor.behavior = 'zoomXY';
      
      let scrollbarX = new am4core.Scrollbar();
      chart.scrollbarX = scrollbarX;

      chart.exporting.menu = new am4core.ExportMenu();
      chart.exporting.menu.verticalAlign = 'bottom';
      chart.exporting.filePrefix = 'dole';

      chart.language.locale = am4lang_es_ES;

      //chart.legend.labels.template.text = "[{color}]{name}[/]";     

      let legendContainer = am4core.create("aquacheck-legend-div", am4core.Container);
      legendContainer.width = am4core.percent(100);
      legendContainer.height = am4core.percent(100);

      chart.legend = new am4charts.Legend();
      chart.legend.parent = legendContainer;

      chart.responsive.enabled = true;

      this._generateAquacheckChartSeries(chart);            

      if(data) {
        chart.data = data;
      }

      this.showGuideControls = true;
      this._chart = chart;

      chart.events.on("datavalidated", _resizeLegend);
      chart.events.on("maxsizechanged", _resizeLegend);
      chart.events.on("ready", _resizeLegend);

      function _resizeLegend(ev) {
        document.getElementById('aquacheck-legend-div').style.height = chart.legend.contentHeight + "px";
      }
  }

  private _generateAquacheckChartSeries(chart: am4charts.XYChart): void {
    const humSerieName: string[] = [
      'Hum 10', 'Hum 20', 'Hum 30', 'Hum 40', 'Hum 60', 'Hum 80', 'Hum Amb',     
    ]
  
    const humSerieApiName: string[] = [
      'hum10Cm', 'hum20Cm', 'hum30Cm', 'hum40Cm', 'hum60Cm', 'hum80Cm', 'humAmb', 
    ]

    const tempSerieName: string[] = [
      'Temp 10', 'Temp 20', 'Temp 30', 'Temp 40', 'Temp 60', 'Temp 80', 'Temp Amb'
    ]

    const tempSerieApiName: string[] = [
      'temp10Cm', 'temp20Cm', 'temp30Cm', 'temp40Cm', 'temp60Cm', 'temp80Cm', 'tempAmb'
    ]

    for(let i = 0; i < humSerieName.length; ++i) {
      let series = chart.series.push(new am4charts.LineSeries());
      series.dataFields.dateX = 'horaToma';
      series.dataFields.valueY = humSerieApiName[i];
      series.name = humSerieName[i];
      series.yAxis = chart.yAxes.getIndex(0);
      series.stroke = chart.colors.getIndex(i);
      series.tensionX = 0.75;
      series.minBulletDistance = 25;
      //series.tooltipText = '{valueY.value}';
      series.legendSettings.valueText = '{valueY}'

      if(humSerieName[i] === 'Hum Amb') {
        series.hidden = true;
        series.stroke = chart.colors.getIndex(20);
      }

      let bullet = series.bullets.push(new am4charts.CircleBullet());
      bullet.circle.strokeWidth = 2;
      bullet.circle.radius = 3;
    }

    for(let i = 0; i < tempSerieName.length; ++i) {
      let series = chart.series.push(new am4charts.LineSeries());
      series.dataFields.dateX = 'horaToma';
      series.dataFields.valueY = tempSerieApiName[i];
      series.name = tempSerieName[i];
      series.yAxis = chart.yAxes.getIndex(1);
      series.stroke = chart.colors.getIndex(i+30);
      series.tensionX = 0.75;
      series.minBulletDistance = 25;
      //series.tooltipText = '{valueY.value}';
      series.legendSettings.valueText = '{valueY}'
      series.hidden = true;

      if(tempSerieName[i] === 'Temp Amb') {
        series.stroke = chart.colors.getIndex(12);
      }

      let bullet = series.bullets.push(new am4charts.CircleBullet());
      bullet.circle.strokeWidth = 2;
      bullet.circle.radius = 3;
    }
  }

  ngOnInit() {
  }

  ngOnDestroy() {
    this.ngZone.runOutsideAngular(() => {
      if (this._chart) {
        this._chart.dispose();
      }
    });
  }
}
```

When chart is generated, console shows this message:

```
TypeError: Cannot read property 'value' of undefined
    at LineSeriesDataItem.push../node_modules/@amcharts/amcharts4/.internal/core/DataItem.js.DataItem.hasValue (DataItem.js:872)
    at LineSeries.push../node_modules/@amcharts/amcharts4/.internal/charts/series/XYSeries.js.XYSeries.positionBullet (XYSeries.js:1018)
    at Series.js:636
    at Iterator.js:57
    at Iterator.js:10
    at Module.each (Iterator.js:56)
    at LineSeries.push../node_modules/@amcharts/amcharts4/.internal/charts/series/Series.js.Series.validateDataElement (Series.js:588)
    at LineSeries.push../node_modules/@amcharts/amcharts4/.internal/core/Component.js.Component.validateDataElements (Component.js:402)
    at LineSeries.push../node_modules/@amcharts/amcharts4/.internal/charts/series/XYSeries.js.XYSeries.hide (XYSeries.js:1205)
    at Component.js:733
```

Could you guide me to solve it?

Thanks in advance. 
Regards.

**Comment by martynasma** [comment]
The code looks OK, so it's hard to say without being able to run the app.

Would you be able to create a working standalone example which we can test?

**Comment by jmpmscorp** [comment]
Do you need an Angular or plain javascript example?

**Comment by jmpmscorp** [comment]
Hi again @martynasma.

I have new news. The error seems to be related with line ```series.hidden = true```. If I keep code as I posted, but I comment this line, chart is shown perfectly again and no error appears.

Because I need some lines hidden at start I have changed line ```series.hidden = true``` by ```series.hide()``` and all work correctly too, no error.

I don't know what is happening with ```series.hidden = true``` in versions 4.2.3+ but using hide() function my app works again. Let me know If I can help you.

**Comment by martynasma** [comment]
Thanks for additional info!

I was able to reproduce in a small demo:
https://codepen.io/team/amcharts/pen/f67c009ca4978cc46166e48cbf9c14d3?editors=0010

Will keep you posted when fix is available.

**Comment by martynasma** [comment]
Fixed in [4.3.5](https://github.com/amcharts/amcharts4/releases/tag/4.3.5).

## [4.3.5] - 2019-03-29

### Added
- New properties `strokeLinejoin` ("miter" | "round" | "bevel") and `strokeLinecap` ("butt" | "square" | "round") added to `Sprite`. [More info](https://developer.mozilla.org/en-US/docs/Web/SVG/Tutorial/Fills_and_Strokes#Stroke).

### Fixed
- In some rare cases EDGE/IE could generate an "Unspecied error".
- `MapChart` method `zoomToMapObject()` was not functioning properly since `4.3.4`.
- `LineSeries` with bullet and `hidden = false` in init was causing the whole chart to break.

[Full change log](https://github.com/amcharts/amcharts4/blob/master/dist/es2015/CHANGELOG.md).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.
